v23.1/ui-key-visualizer.md

```diff
@@ -0,0 +1,103 @@
+
+To access the key visualizer, [navigate to the DB Console](ui-overview.html#db-console-access) and click **Advanced Debug** in the left-hand navigation, then click **Key Visualizer** under the **Even More Advanced Debugging** heading.
+
+## Required privileges
+
+To view the **Advanced Debug** page, and work with the key visualizer, the user must be a member of the `admin` role or must have the `VIEWDEBUG` [system privilege](security-reference/authorization.html#supported-privileges) defined.
+
+## Enable the key visualizer
+
+To use the key visualizer, the `keyvisualizer.enabled` [cluster setting](cluster-settings.html) must be set to `true`, using the [`SET CLUSTER SETTING`](set-cluster-setting.html) statement:
+
+~~~ sql
+> SET CLUSTER SETTING keyvisualizer.enabled = true;
+~~~
+
+The key visualizer is disabled by default.
+
+## Key visualizer features
+
+Once you have enabled the key visualizer, CockroachDB will begin monitoring keyspace usage across your cluster. Data is only collected while the key visualizer is enabled; you cannot visualize data from a time period before you enabled it.
+
+When navigating to the **Key Visualizer** page in the DB Console, after a brief loading time, CockroachDB will present the collected data in a visualization designed to help you see data traffic trends at a glance.
+
+<img src="{{ 'images/v23.1/key-visualizer-hot-range.png' | relative_url }}" alt="Key visualizer showing hot range" style="max-width:80%" />
+
+The key visualizer presents the following information:
+
+- The entire [table data keyspace](architecture/distribution-layer.html#table-data-kv-structure) is represented on the Y-axis, and is broken up into buckets representing whole ranges of the keyspace, or aggregated ranges if the number of ranges in the keyspace exceeds the [configured number of buckets](#key-visualizer-customization).
+
+- Time is represented on the X-axis, with its granularity (i.e., frequency of data collection) being controlled by the [configured sample period](#key-visualizer-customization).
+
+- Keyspace activity is visualized on a color scale from black to red, representing "cold" and "hot" respectively. Thus, a range shown in bright red indicates a [hot spot](performance-best-practices-overview.html#hot-spots), while a range shown in black indicates a range with little to no active reads or writes. Hot spots are identified relative to other ranges, e.g., a range that receives one write a minute could be considered a hot spot if the rest of the ranges on the cluster aren't receiving any. A range shown in red is therefore not necessarily itself indicative of a problem, but a useful visual on the path to troubleshooting one.
```

Comment:
I felt that the last sentence could be more clear—the last clause seems to assume (depending on interpretation) that there _is_ a problem to troubleshoot, but the first clause raises the possibility that this isn't the case. Here's just one potential way to make it more clear, if you agree with the message.

```suggestion
- Keyspace activity is visualized on a color scale from black to red, representing "cold" and "hot" respectively. Thus, a range shown in bright red indicates a [hot spot](performance-best-practices-overview.html#hot-spots), while a range shown in black indicates a range with little to no active reads or writes. Hot spots are identified relative to other ranges; for example, a range that receives one write per minute could be considered a hot spot if the rest of the ranges on the cluster aren't receiving any. A range shown in red is therefore not necessarily indicative of a problem, but it may call attention to an area that needs attention to prevent a problem.
```

Reply:
> it may call attention to an area that needs attention

Oops on the word repetition! Not my best work, but hopefully the suggestion got my point across.

Reply:
Missed it in my initial review of your review. Thank you for reviewing your review! (caught and pushed a fix, should be ready to merge after this).